shotgun_api3/shotgun.py

```diff
@@ -4470,6 +4470,28 @@ def _translate_filters_simple(sg_filter):
     if len(values) == 1 and isinstance(values[0], (list, tuple)):
         values = values[0]
 
+    # Payload optimization: Do not send a full object
+    # just send the `type` and `id` when combining related queries
+    disable_entity_optimization = os.environ.get(
+        "SHOTGUN_API_DISABLE_ENTITY_OPTIMIZATION"
```

Comment:
Wouldn't it be better if it were dynamic? For instance, if the query only includes the `id` and `type `fields, it could be made with just that data, without relying on the environment variable. Because what happens if I forget to deactivate the environment variable and use more parameters than just `id` and `type`?

Reply:
The point of having this environmental variable is only to "disable" this new feature and behave as it was. It can help ups detect any potential issues while leaving a quick fix for the testers.